gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -266,6 +266,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                 kspTaskProvider = kspTaskProvider,
                 javaOutputDir = javaOutputDir,
                 classOutputDir = classOutputDir,
+                resourcesOutputDir = project.fileTree(resourceOutputDir)
```

Comment:
You want to add the top-level dir only, so using `project.files(resourceOutputDir)` should do the trick. This is also causing the test failure as AGP assumes only jars and top-level dirs are added.

Reply:
9b07458